src/main/java/com/microsoft/jenkins/appservice/commands/FileDeployCommand.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.jenkins.appservice.commands;
+
+import com.microsoft.azure.management.appservice.DeploymentSlot;
+import com.microsoft.azure.management.appservice.WebApp;
+import com.microsoft.jenkins.appservice.AzureAppServicePlugin;
+import com.microsoft.jenkins.appservice.util.Constants;
+import com.microsoft.jenkins.appservice.util.DeployTypeEnum;
+import com.microsoft.jenkins.azurecommons.command.CommandState;
+import com.microsoft.jenkins.azurecommons.command.IBaseCommandData;
+import com.microsoft.jenkins.azurecommons.command.ICommand;
+import com.microsoft.jenkins.azurecommons.telemetry.AppInsightsUtils;
+import hudson.FilePath;
+import hudson.Util;
+import org.apache.commons.lang.StringUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileDeployCommand implements ICommand<FileDeployCommand.IWarDeployCommandData> {
+    private String aiDeployInfo;
+    private String aiDeployFailedInfo;
+
+    @Override
+    public void execute(IWarDeployCommandData context) {
+        WebApp app = context.getWebApp();
+        DeployTypeEnum deployType = context.getDeployType();
+
+        FilePath srcDir = context.getJobContext().getWorkspace().child(Util.fixNull(context.getSourceDirectory()));
+        try {
+            FilePath[] files = srcDir.list(Util.fixNull(context.getFilePath()));
+
+            switch (deployType) {
+                case ZIP:
+                    aiDeployInfo = Constants.AI_ZIP_DEPLOY;
+                    aiDeployFailedInfo = Constants.AI_ZIP_DEPLOY_FAILED;
+                    List<FilePath> zipFilesPathList = new ArrayList<>();
+                    for (FilePath file : files) {
+                        if (!file.getName().toLowerCase().endsWith(Constants.ZIP_FILE_EXTENSION)) {
+                            context.logStatus(file.getName() + " is not ZIP file. Will skip.");
+                            continue;
+                        }
+                        zipFilesPathList.add(file);
+                    }
+                    if (zipFilesPathList.isEmpty()) {
+                        context.logStatus("Choose ZIP deployment, but there is no zip files in your "
+                                + "setting workspace.");
+                    }
+                    if (zipFilesPathList.size() > 1) {
```

Comment:
How about making the zip package for user?

Reply:
How to make the zip package for user? Package all the files in a folder?

Reply:
If packaging zip for user, should there be a choice for user to choose what to pack in? what's  the normal user flow without Jenkins? user package all things under build folder directly?

Reply:
In Git deploy, user can specify file name pattern like `**/*.js,**/*.css` to choose what to check in. Zip deploy can be similar.